Libplanet/KeyStore/ProtectedPrivateKey.cs

```diff
@@ -0,0 +1,402 @@
+        /// <param name="cipher">A symmetric cipher to decrypt a <see cref="PrivateKey"/>.</param>
+        /// <param name="ciphertext">An encrypted <see cref="PrivateKey"/>.</param>
+        public ProtectedPrivateKey(
+            Address address,
+            IKdf kdf,
+            byte[] mac,
+            ICipher cipher,
+            byte[] ciphertext
+        )
+            : this(
+                address,
+                kdf,
+                ImmutableArray.Create(mac),
+                cipher,
+                ImmutableArray.Create(ciphertext)
+            )
+        {
+        }
+
+        /// <summary>
+        /// Loads a protected private key.
+        /// </summary>
+        /// <param name="address">The address of the protected private key.</param>
+        /// <param name="kdf">A key derivation function to derive a symmetric key to decrypt
+        /// a <see cref="PrivateKey"/>.</param>
+        /// <param name="mac">MAC digest to check if a derived key is correct or not.</param>
+        /// <param name="cipher">A symmetric cipher to decrypt a <see cref="PrivateKey"/>.</param>
+        /// <param name="ciphertext">An encrypted <see cref="PrivateKey"/>.</param>
+        public ProtectedPrivateKey(
+            Address address,
+            IKdf kdf,
+            ImmutableArray<byte> mac,
+            ICipher cipher,
+            ImmutableArray<byte> ciphertext
+        )
+        {
+            Address = address;
+            Kdf = kdf;
+            Mac = mac;
+            Cipher = cipher;
+            Ciphertext = ciphertext;
+        }
+
+        /// <summary>
+        /// The address of the protected private key.
+        /// </summary>
+        public Address Address { get; }
+
+        /// <summary>
+        /// A key derivation function to derive a symmetric key to decrypt
+        /// a <see cref="PrivateKey"/>.
+        /// </summary>
+        [Pure]
+        public IKdf Kdf { get; }
+
+        [Pure]
+        public ImmutableArray<byte> Mac { get; }
+
+        /// <summary>
+        /// A symmetric cipher to decrypt a <see cref="PrivateKey"/>.
+        /// </summary>
+        [Pure]
+        public ICipher Cipher { get; }
+
+        /// <summary>
+        /// An encrypted <see cref="PrivateKey"/>.
+        /// </summary>
+        [Pure]
+        public ImmutableArray<byte> Ciphertext { get; }
+
+        /// <summary>
+        /// Protects a bare <see cref="PrivateKey"/> using a user input
+        /// <paramref name="passphrase"/>.
+        /// </summary>
+        /// <param name="privateKey">A bare private key to protect.</param>
+        /// <param name="passphrase">A user input passphrase (i.e., password).</param>
+        /// <returns>A passphrase-protected private key.</returns>
+        [Pure]
+        public static ProtectedPrivateKey Protect(PrivateKey privateKey, string passphrase)
+        {
+            var salt = new byte[32];
+            var rng = RandomNumberGenerator.Create();
```

Comment:
We can use `using` statement against `RandomNumberGenerator` because it's `IDisposable`.

Reply:
Fixed: 67c50509abacfe3dfb376638dc7b4aab8169b3a3.